Content.Shared/_Impstation/Kodepiia/KodepiiaConsumedSystem.cs

```diff
@@ -0,0 +1,33 @@
+using Content.Shared._Impstation.Kodepiia.Components;
+using Content.Shared.Examine;
+using Content.Shared.IdentityManagement;
+using Content.Shared.Mobs;
+
+namespace Content.Shared._Impstation.Kodepiia;
+
+public sealed class KodepiiaConsumedSystem : EntitySystem
+{
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<KodepiiaConsumedComponent, ExaminedEvent>(OnExamine);
+        SubscribeLocalEvent<KodepiiaConsumedComponent, MobStateChangedEvent>(OnMobStateChange);
+    }
+    private void OnExamine(Entity<KodepiiaConsumedComponent> ent, ref ExaminedEvent args)
+    {
+        var locIndex = ent.Comp.TimesConsumed switch
+        {
+            >= 8 => 4,
+            >= 4 => 3,
+            >= 2 => 2,
+            _ => 1,
+        };
+        args.PushMarkup(Loc.GetString($"kodepiia-consumed-onexamine-{locIndex}", ("target", Identity.Entity(ent, EntityManager))));
+    }
+    private void OnMobStateChange(Entity<KodepiiaConsumedComponent> ent, ref MobStateChangedEvent args)
+    {
+        if (args.NewMobState != MobState.Dead)
+            RemComp<KodepiiaConsumedComponent>(ent);
```

Comment:
full "regeneration" after revival feels kinda strange but idk a better way of handling it. i guess rotting does the same shit. space magic

Reply:
yeah